### PR TITLE
Release Preparation Work

### PR DESCRIPTION
**Pull request by cassava**
This pull-request bundles a bunch of commits that need to be merged before we can create an Eclipse release.

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def set_version(self):
     def requirements(self):
         self.requires("boost/[~=1.69]", override=True)
         self.requires("fmt/[~=8.1.1]")
+        self.requires("inja/[~=3.3.0]")
```
Comment (tobifalk) [inline conanfile.py:40]: Why don't we use `override=True` here since `inja` is not a direct dependency?

File: conanfile.py [hunk 2]
```diff
@@ -38,6 +38,7 @@ def requirements(self):
         self.requires("boost/[~=1.69]", override=True)
         self.requires("fmt/[~=8.1.1]")
         self.requires("inja/[~=3.3.0]")
+        self.requires("nlohmann_json/[~=3.10.5]")
```
Comment (tobifalk) [inline conanfile.py:41]: Why don't we use `override=True` here since `nlohmann_json` is not a direct dependency?

File: conanfile.py [hunk 3]
```diff
@@ -39,6 +39,7 @@ def requirements(self):
         self.requires("fmt/[~=8.1.1]")
         self.requires("inja/[~=3.3.0]")
         self.requires("nlohmann_json/[~=3.10.5]")
+        self.requires("incbin/[~=0.88.0]@cloe/stable"),
```
Comment (tobifalk) [inline conanfile.py:42]: Why don't we use `override=True` here since `incbin` is not a direct dependency?

File: runtime/conanfile.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ class CloeRuntime(ConanFile):
         "boost/[>=1.65.1]",
         "inja/[~=3.3.0]",
         "spdlog/[~=1.9.0]",
-        "incbin/[~=1.74.0]@cloe/stable",
+        "incbin/[~=0.88.0]@cloe/stable",
```
Comment (tobifalk) [inline runtime/conanfile.py:37]: What's the reason for this downgrade?

Comment (cassava) [reply runtime/conanfile.py:37]: I think I explained in the commit message.
It goes like this:

1. incbin doesn't version
2. We invent some versioning based on the number of commits
3. We originally did `1.{#commits}.{#our-changes}`
4. The starting `1` is a bad idea though, if incbin every starts versioning itself.
5. So we drop the major version to 0.

File: conanfile.py [hunk 5]
```diff
@@ -36,6 +36,7 @@ def set_version(self):
 
     def requirements(self):
         self.requires("boost/[~=1.69]", override=True)
+        self.requires("fmt/[~=8.1.1]")
```
[the remark this answers is not in the document]
Comment (tobifalk) [reply conanfile.py:39]: At least in the [Conan documentation](https://docs.conan.io/en/1.45/using_packages/conanfile_txt.html#overriding-requirements) they frequently talk about `override` in context of transitive dependencies.

File: runtime/conanfile.py (hunk 4 above)
Comment (cassava) [reply runtime/conanfile.py:37]: So it's not a downgrade, but a re-versioning.

**Review by tobifalk — APPROVED** [review]
LGTM!

**Pull request merged**